libs/community/langchain_community/embeddings/bedrock.py

```diff
@@ -112,40 +156,37 @@ def validate_environment(cls, values: Dict) -> Dict:
 
         return values
 
-    def _embedding_func(self, text: str) -> List[float]:
+    def _embedding_func(
+        self, texts: List[str], input_type: Optional[str] = None
+    ) -> List[List[float]]:
         """Call out to Bedrock embedding endpoint."""
         # replace newlines, which can negatively affect performance.
-        text = text.replace(os.linesep, " ")
+        texts = [text.replace("\n", " ") for text in texts]
 
         # format input body for provider
         provider = self.model_id.split(".")[0]
         _model_kwargs = self.model_kwargs or {}
-        input_body = {**_model_kwargs}
-        if provider == "cohere":
-            if "input_type" not in input_body.keys():
-                input_body["input_type"] = "search_document"
-            input_body["texts"] = [text]
-        else:
-            # includes common provider == "amazon"
-            input_body["inputText"] = text
-        body = json.dumps(input_body)
+        input_bodies = EmbeddingInputOutputAdapter.prepare_inputs(
+            provider, texts, _model_kwargs, input_type
+        )
 
         try:
             # invoke bedrock API
-            response = self.client.invoke_model(
-                body=body,
-                modelId=self.model_id,
-                accept="application/json",
-                contentType="application/json",
-            )
-
-            # format output based on provider
-            response_body = json.loads(response.get("body").read())
-            if provider == "cohere":
-                return response_body.get("embeddings")[0]
-            else:
-                # includes common provider == "amazon"
-                return response_body.get("embedding")
+            responses = [
+                self.client.invoke_model(
+                    body=json.dumps(input_body),
+                    modelId=self.model_id,
+                    accept="application/json",
+                    contentType="application/json",
+                )
+                for input_body in input_bodies
+            ]
+
+            response_bodies = [
+                EmbeddingInputOutputAdapter.prepare_outputs(provider, response)
+                for response in responses
+            ]
```

Comment:
This can be improved through parallelization. As it stands Titan models would observe slowdown. I am open to suggestions.

Reply:
Why would Titan models observe slowdown?

Reply:
By slowdown I mean sequential invocation can take quite long (as there is no batching).

Previous code had an asyncio gather to speed up (which was only available if invoked with async); it was not available for normal embed. So wondering if I should bring that back or there is some better alternative.

Reply:
Another thing that can potentially bite are the rate limits. Not sure what the exact values are for non provisioned instances, but once you start hitting might take hours to go away.

Reply:
Still don't get it. async calls, threads or sequential, they all do the same things from the API's point of view and hence are equivalent in terms of throttling except that with concurrent requests (async / threads) you'll consume your "quota" faster. Also I think the throttling should happen on a per minute level and should clear up after 60 seconds. Unless it works differently for embeddings than it does for LLM calls.

Reply:
Resolving this conversation. We have not observed any throttles in our use cases for embeddings, and if there is any issue happening in the wild a separate issue regarding such may be opened.